assets/js/components/settings/settings-modules.js

```diff
@@ -82,31 +81,16 @@ class SettingsModules extends Component {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param {string} module The module slug.
-	 * @param {string} action The action being performed, one of 'edit', 'cancel' or 'confirm'.
+	 * @param {string}   module        The module slug.
+	 * @param {string}   action        The action being performed, one of 'edit', 'cancel' or 'confirm'.
+	 * @param {Function} submitChanges The action dispatcher to submit the changes.
 	 */
-	handleButtonAction( module, action ) {
+	async handleButtonAction( module, action, submitChanges ) {
 		if ( 'confirm' === action ) {
-			const modulePromise = applyFilters( 'googlekit.SettingsConfirmed', false, module );
-
 			this.setState( { isSaving: module } );
-			if ( ! modulePromise ) {
-				// Clears session and local storage on successful setting.
-				clearWebStorage();
-
-				return;
-			}
-			modulePromise.then( () => {
-				// Clears session and local storage on every successful setting.
-				clearWebStorage();
-
-				this.setState( {
-					isSaving: false,
-					error: false,
-				} );
+			const { error: err } = await submitChanges( module );
```

Comment:
Why define this as `err`? `error` is a fine variable name here and means the code will be a bit less complicated 🙂